olive/engine/engine.py

```diff
@@ -43,52 +43,55 @@ class Engine:
 
     def __init__(
         self,
-        config: Union[Dict[str, Any], EngineConfig] = None,
-        search_strategy: Optional[SearchStrategy] = None,
-        host_config: Optional[SystemConfig] = None,
-        target_config: Optional[SystemConfig] = None,
-        evaluator_config: Optional["OliveEvaluatorConfig"] = None,
+        search_strategy: Optional[Union[Dict[str, Any], SearchStrategy]] = None,
```

Comment:
Could you add `SearchStrategyConfig` as another option here?

to make it consistent with the other parameters, we could also remove `SearchStrategy` from accepted type and also remove the `else` clause at line 65.